### PR TITLE
Problem: client-rpc doen't flush sled database, possibly causes lost data (fix #2100)

### LINKED ISSUE #2100: Problem: client-rpc doen't flush sled database, possibly causes lost data
In recent sled database, it doesn't flush automatically,
in normal shutdown, `drop` is called, so OK

but like client-rpc, not graceful close,
some data will be lost
so 
1. call flush after dbcalls
2. add polling function which does flushing periodically

### PR DESCRIPTION
**Pull request by leejw51crypto**
Solution: add flushing

in recent sled, flushing is not working as previous version.
flushes on drop only.
so added manual flushing to `db writing` functions
(sled Config.flush_every_ms didn't work)

### CODE REVIEW

**Comment by leejw51crypto** [comment]
ok, i'll check clippy issue

**Comment by leejw51crypto** [comment]
testing sled 0.34.2

File: client-rpc/src/handler.rs [hunk 1]
```diff
@@ -49,6 +49,16 @@ impl RpcHandler {
...
         let storage = SledStorage::new(&storage_dir)?;
+
+        let polling_storage = storage.clone();
+        std::thread::spawn(move || {
+            loop {
+                polling_storage.flush().expect("sled storage flush");
+                // every 1 second
+                std::thread::sleep(std::time::Duration::from_secs(1));
+            }
+        });
```
Comment (tomtau) [inline client-rpc/src/handler.rs:60]: sled has this config: https://github.com/spacejam/sled/blob/master/src/config.rs#L394

that doesn't work?

Comment (tomtau) [reply client-rpc/src/handler.rs:60]: Oh I see the commit comment
> (sled Config.flush_every_ms didn't work)

Comment (leejw51crypto) [reply client-rpc/src/handler.rs:60]: strange, that should work, i'll try again

**Comment by leejw51crypto** [comment]
added `flushing periodically every 1 second`,
so if , flushing is not done,  data will be written safely to the disk

**Comment by tomtau** [comment]
bors try

**Comment by tomtau** [comment]
@leejw51crypto can you change the commit message to reflect the updated issue name?

File: client-core/src/wallet/syncer.rs [hunk 2]
```diff
@@ -597,6 +597,7 @@ impl<
                 self.sync_state.last_block_hash = block.block_hash.clone();
 
                 self.update_progress(block.block_height);
+                self.env.storage.flush()?;
```
Comment (yihuang) [inline client-core/src/wallet/syncer.rs:600]: Should it be put in the `save` method?

Comment (leejw51crypto) [reply client-core/src/wallet/syncer.rs:600]: looking good, i'll check

**Comment by leejw51crypto** [comment]
i'll check integration test

**Comment by tomtau** [comment]
```
Traceback (most recent call last):
630	  File "./multinode/reward_test.py", line 45, in <module>
631	    assert int(state['bonded']) == init_bonded + minted // 2
632	AssertionError
```

it seems it may be some timing issue in the rewards test ? @yihuang

**Comment by tomtau** [comment]
bors retry

**Comment by tomtau** [comment]
bors retry

**Pull request merged**